templates/vhost/vhost_header.erb

```diff
@@ -94,6 +94,16 @@ server {
   passenger_set_cgi_param  <%= key %> <%= @passenger_cgi_param[key] %>;
   <%- end -%>
 <% end -%>
+<% if @passenger_header -%>
+  <%- @passenger_header.keys.sort.each do |key| -%>
+  passenger_set_header  <%= key %> <%= @passenger_header[key] %>;
+  <%- end -%>
+<% end -%>
+<% if @passenger_env_var -%>
+  <%- @passenger_env_var.keys.sort.each do |key| -%>
+  passenger_set_env_var  <%= key %> <%= @passenger_env_var[key] %>;
```

Comment:
I was following the pattern for 'passenger_set_cgi_param' which only uses '_set_' in the template when it is actually writing the config.